xbmc/filesystem/File.cpp

```diff
@@ -1080,7 +1080,7 @@ CFileStreamBuffer::pos_type CFileStreamBuffer::seekoff(
 
   int64_t position = -1;
   if(way == std::ios_base::cur)
-    position = m_file->Seek(offset, SEEK_CUR);
+    position = m_file->Seek(pos+offset, SEEK_SET); // Dont seek from current pos, because m_file does readahead and so has wrong position
```

Comment:
why don't you correct offset by readahead bytes? there may be a reason why clients call this with SEEK_CUR